source/flow.rst

```diff
@@ -10,4 +10,303 @@ and events generated by devices.
+    :width: 50%
+    :align: center
+    :alt: switch_node_cfg
+
+    : Switch configuration
+
+Fields:
+
+* **Name** *(optional)*: Name of the node
+* **Property** *(required)*: Variable that will be evaluated 
+* **Rule box** *(required)*: Rules that will determine the outputch branch of the node.
+  Also, it can be configured to stop checking rules when it finds one that matches other
+  or check all the rules and route the message to the corresponding output.
+
+.. note::
+    - More than one rule can be assign by clicking on *+add* below the rules box.
+    - Each condition should have an corresponding output.
+
+Template
+********
+
+.. note::
+    Despite the name, this node has nothing to do with dojot templates
+
+.. _template_node:
+.. image:: images/nodes/template_node.png
+    :width: 20%
+    :align: left
+    :alt: template_node
+
+This node will assign a value to a target variable. This value can be a constant,
+the value of an attribute that came from the entry device and etc...
+
+It uses the `mustache`_ template language.
+Check :numref:`template_node_cfg` as example:
+the field **a** of payload will be replaced with the value of the **payload.b**
+
+
+
+.. _template_node_cfg:
+.. figure:: images/nodes/template_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: template_node_cfg
+
+Fields:
+
+* **Name** *(optional)*: Name of the node
+* **Set Property** *(required)*: Variable that will receive the value
+* **Format** *(required)*: Format template will be writen
+* **Template** *(required)*: Value that will be assigned to the target variable set on **Set property**
+* **Output as** *(required)*: The format of the output
+
+Email
+*****
+
+.. _email_node:
+.. image:: images/nodes/email_node.png
+    :width: 20%
+    :align: left
+    :alt: email_node
+
+Sends an e-mail for a given address.
+
+.. _email_node_cfg:
+.. figure:: images/nodes/email_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: email_node_cfg
+
+Fields:
+
+* **From** *(required)*: The source email.
+* **To** *(required)*: Destination email.
+* **Server** *(required)*: The server of the email destination.
+* **Subject** *(required)*: Subject of the email.
+* **Body** *(required)*: Message on the email. The message can be writen in a variable using the **template node**.
+* **Name** *(optional)*: Name of the node.
+
+Geofence
+********
+
+.. _geofence_node:
+.. image:: images/nodes/geofence_node.png
+    :width: 20%
+    :align: left
+    :alt: geofence_node
+
+Select an interest area to determine wich devices will actuate the flow
+
+.. _geofence_node_cfg:
+.. figure:: images/nodes/geofence_node_cfg.png
+    :width: 50%
+    :align: center
+    :alt: geofence_node_cfg
+
+Fields:
+
+* **Area** *(required)*: Area that will be selected. It can be selected with an square or with a pentagon.
+* **Filter** *(required)*: Select wich side of the area will be selected: inside or outside the area demercated in the field above.
```

Comment:
> Select wich side of the area...

Select which side of the area...

> the area demercated...

the marked area?

BTW, there's too much 'selected'. Might be good to rephrase this item.